File: backend/schemas/retention.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class RetentionPolicyResponse(BaseModel):
    """Static, cacheable data-retention policy metadata (issue #43, plan §5.3).

    Served unauthenticated from ``GET /retention/policy`` so the delete dialog,
    the "Recently deleted" section, and the Settings "Data retention" panel all
    render the live windows from one source of truth instead of hard-coding them.
    ``policy_version`` is the string the delete dialog stamps as
    ``retention_ack_version`` on confirm (proof the user saw the notice).
    """

    policy_version: str
    # Tier-3 workspace trash windows (days).
    trash_days: int
    legacy_archived_days: int
    # Tier-2 content keep-N.
    stage_versions_keep: int
    stage_versions_min_age_days: int
    storyboards_keep: int
    storyboards_min_age_days: int
    # Tier-1 telemetry TTL windows (days) — surfaced for the Settings panel.
    cost_events_days: int
    eval_results_days: int

    model_config = ConfigDict(extra="forbid")

File: frontend/src/types/retention.ts
import type { AIProvider } from "./workspace"

/**
 * Static data-retention policy metadata from GET /retention/policy (issue #43).
 * The delete dialog stamps `policy_version` as the ack; the Settings panel and
 * the "Recently deleted" countdown render the windows from this one source.
 */
export interface RetentionPolicy {
  policy_version: string
  trash_days: number
  legacy_archived_days: number
  stage_versions_keep: number
  stage_versions_min_age_days: number
  storyboards_keep: number
  storyboards_min_age_days: number
  cost_events_days: number
  eval_results_days: number
}

/** A trashed (archived) workspace from GET /workspaces/trashed (issue #43). */
export interface TrashedWorkspace {
  id: string
  name: string
  provider: AIProvider
  archived_at: string
  purge_after: string
  acknowledged: boolean
}
